### DISCUSSION TITLE
给角色分配权限、可正常增加权限、但减少权限不正常

### DISCUSSION THREAD

**Discussion by 18285160741** [zuihou/zuihou-admin-cloud#86] [02.提问]
【有劳指点】给角色分配权限、可正常增加权限、但减少权限不正常。

第一步：查看、角色对应的菜单是这样的
![WKNW`LEVPU47Z3FQD4UO~FX](https://user-images.githubusercontent.com/37412616/113975816-ad880b80-9872-11eb-80ad-6e7e6b13eb31.jpg)

第二步：减少菜单、提交
![L88PC4NCS($%5037_1)L `B](https://user-images.githubusercontent.com/37412616/113975928-d3151500-9872-11eb-9b52-86ae87d7cdc0.jpg)

第三步：再次查看、角色对应的菜单是这样的。明明去掉的菜单又显示出来了
![{X63E922TCJVUKPFT`FMSVY](https://user-images.githubusercontent.com/37412616/113976091-166f8380-9873-11eb-9c1c-b16fd724a410.jpg)

第四步：增加菜单、提交。再次查看：显示正常
![T1U1{O0UN5JO~CI@Y%CZ2MM](https://user-images.githubusercontent.com/37412616/113976203-52a2e400-9873-11eb-9848-b3a509adbd89.png)

**Comment by zuihou** [comment]
这里应该是前端有bug

**Reply by 18285160741** [reply]
我知道了。是这段代码的问题。注释掉就好了（不应该按资源找菜单、因为前端提交了很多没有对应菜单的resourceId）。可能存在API安全风险
![image](https://user-images.githubusercontent.com/37412616/113978489-95b28680-9876-11eb-9d8b-a1a44ac1a1c7.png)

**Comment by 18285160741** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
